File: easy/TwoNumberSum/Python/solution1.py
def twoNumberSum(array, targetSum):
    # Write your code here.
	# Time: O(n) | Space: O(1)
	array.sort()
	left = 0
	right = len(array) - 1
	while (left < right):
		possibleSum = array[left] + array[right]
		if (possibleSum == targetSum):
			return [array[left], array[right]]
		elif (possibleSum < targetSum):
			left += 1
		elif (possibleSum > targetSum):
			right -=1
	return []

File: easy/TwoNumberSum/Javascript/solution1.js
function twoNumberSum(array, targetSum) {
	const nums = {};
	for (const num of array) {
		const complement = targetSum - num;
		if (complement in nums) {
			return complement > num ? [num, complement] : [complement, num];
		} else {
			nums[num] = true;
		}
	}
	return [];
}

